From 0072b3f21ec811c2eee3f9c6f386c84952d39e31 Mon Sep 17 00:00:00 2001
From: Michal Sekletar <msekleta@redhat.com>
Date: Mon, 31 Oct 2016 17:26:31 +0100
Subject: [PATCH] avahi-dnsconfd initscript improvements

---
 initscript/fedora/avahi-dnsconfd.in | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/initscript/fedora/avahi-dnsconfd.in b/initscript/fedora/avahi-dnsconfd.in
index bd4182e..da637c0 100644
--- a/initscript/fedora/avahi-dnsconfd.in
+++ b/initscript/fedora/avahi-dnsconfd.in
@@ -49,6 +49,7 @@ start() {
     # Check that networking is configured.
     [ ${NETWORKING} = "no" ] && exit 1
 
+	[ "$EUID" != "0" ] && exit 4
 	echo -n $"Starting Avahi DNS daemon... "
         daemon --pidfile=${PIDFILE} $AVAHI_BIN $AVAHI_OPTS
         RETVAL=$?
@@ -58,6 +59,7 @@ start() {
 }
 
 stop() {
+    	[ "$EUID" != "0" ] && exit 4
         echo -n $"Shutting down Avahi DNS daemon: "
         killproc -p ${PIDFILE} $AVAHI_BIN
         RETVAL=$?
@@ -75,6 +77,7 @@ reload() {
 }
 
 restart() {
+	[ "$EUID" != "0" ] && exit 4
 	stop
 	start
 }
@@ -96,16 +99,16 @@ case "$1" in
   restart)
 	restart
 	;;
-  reload)
+  reload|force-reload)
         reload
 	;;
-  condrestart)
+  condrestart|try-restart)
   	if [ -f $LOCKFILE ]; then
 		restart
 	fi
 	;;
   *)
-	echo $"Usage: $0 {start|stop|status|restart|condrestart}"
+	echo $"Usage: $0 {start|stop|status|restart|condrestart|try-restart|reload|force-reload}"
 	exit 2
 	;;
 esac
-- 
2.7.4

