From a0b1523326b5c9014a6ea31854f08cea2d737572 Mon Sep 17 00:00:00 2001
From: Michal Sekletar <msekleta@redhat.com>
Date: Mon, 31 Oct 2016 17:14:41 +0100
Subject: [PATCH] avahi-daemon initscript improvements

---
 initscript/fedora/avahi-daemon.in | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/initscript/fedora/avahi-daemon.in b/initscript/fedora/avahi-daemon.in
index 5a8f430..36a9156 100644
--- a/initscript/fedora/avahi-daemon.in
+++ b/initscript/fedora/avahi-daemon.in
@@ -49,6 +49,7 @@ start() {
     # Check that networking is configured.
     [ ${NETWORKING} = "no" ] && exit 1
 
+	[ "$EUID" != "0" ] && exit 4
 	echo -n $"Starting Avahi daemon... "
 	if [ -s /etc/localtime ]; then
 	    cp -fp /etc/localtime /etc/avahi/etc >/dev/null 2>&1
@@ -61,6 +62,7 @@ start() {
 }
 
 stop() {
+    	[ "$EUID" != "0" ] && exit 4
         echo -n $"Shutting down Avahi daemon: "
         killproc -p ${PIDFILE} $AVAHI_BIN
         RETVAL=$?
@@ -78,6 +80,7 @@ reload() {
 }
 
 restart() {
+	[ "$EUID" != "0" ] && exit 4
 	stop
 	start
 }
@@ -99,16 +102,16 @@ case "$1" in
   restart)
 	restart
 	;;
-  reload)
+  reload|force-reload)
         reload
 	;;
-  condrestart)
+  condrestart|try-restart)
   	if [ -f $LOCKFILE ]; then
 		restart
 	fi
 	;;
   *)
-	echo $"Usage: $0 {start|stop|status|restart|condrestart}"
+	echo $"Usage: $0 {start|stop|status|restart|condrestart|try-restart|reload|force-reload}"
 	exit 2
         ;;
 esac
-- 
2.7.4

